### ISSUE TITLE
Feedback on construction order tracking for repos

### ISSUE THREAD

**Issue by timaeudg** [PolicyStat/gitlit#18]
Below are my findings from the table-based vs. name-based construction order that was from Sprint 1. Be sure to leave some comments on what you think about my findings and/or what you think about each method
### Lexicographic

**Pros:**
- Git diff can help with noting when nodes move
- Joining is super easy, just construct the files and paste them together in lexo order

**Cons:**
- There is more to noting moves than just a git diff:
  - If the starting node gets moved to the end, all the nodes will have different names, which would be a major pain to walk through, since how do we know what got moved?
  - We could use the GUID, or something that gave a clue to the content of the file, but that doesn't really help us, as we would then just need to keep track of what the old ordering and GUIDs were anyway and try to match up with them to prevent unnecessary changes.
  - We could just intelligently create names the minimize file renames, but this requires complex logic.
- Adding new sections requires more complex detection logic than for tables.
- Limited size between pre-existing nodes.
- Usefulness is limited if we don't try to match old filenames.
### Table-Based

**Pros:**
- Git diff would show section changes perfectly
- Filenames could stay the same easily.
- Addition new sections is super easy, just a new file, and a change to the table
- Unlimited size between nodes (actually, I guess the limit is how long we allow the metadata file to get)
- Move detection is just checking the table to see whether the order changed from the old order. Still some work, but much less

**Cons:**
- Would need to do some work for diffs, as we would need to look at the order table to make note of what, if anything, moved
- Joining would require looking at the metadata file
- Need to keep the filenames consistent between document builds (that is, need a consistent way to generate filenames)
### TL;DR

Basically, Table-based makes diffs a bit nicer, and is easy, but does require a bit of work for diffs to be readily usable. That being said, Lexicographic ordering would require even more work than this.
